app/javascript/tests/close_contact/CloseContact.test.js

```diff
@@ -0,0 +1,219 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme';
+import { Button, Modal } from 'react-bootstrap';
+import InfoTooltip from '../../components/util/InfoTooltip';
+import CloseContact from '../../components/close_contact/CloseContact.js'
+import { mockPatient1 } from '../mocks/mockPatients'
+import { mockCloseContact1, mockCloseContact2 } from '../mocks/mockCloseContact'
+
+const authyToken = 'Q1z4yZXLdN+tZod6dBSIlMbZ3yWAUFdY44U06QWffEP76nx1WGMHIz8rYxEUZsl9sspS3ePF2ZNmSue8wFpJGg==';
+const ASSIGNED_USERS = [ 123234, 512678, 910132 ]
+const testInputValues = [
+  { field: 'first_name', value: 'Anthony', inputType: 'FormControl' },
+  { field: 'last_name', value: 'Stark', inputType: 'FormControl' },
+  { field: 'primary_telephone', value: '1234567890', inputType: 'PhoneInput' },
+  { field: 'email', value: 'tinman@starkindustries.com', inputType: 'FormControl' },
+  { field: 'last_date_of_exposure', value: '2020-05-16', inputType: 'DateInput' },
+  { field: 'assigned_user', value: ASSIGNED_USERS[0], inputType: 'FormControl' },
+  { field: 'notes', value: 'Inevitable', inputType: 'FormControl' },
+]
+
+function getShallowWrapper(closeContact) {
+  return shallow(<CloseContact authenticity_token={authyToken} patient={mockPatient1} close_contact={closeContact}
+    can_enroll_patient_close_contacts={true} assigned_users={ASSIGNED_USERS}  />);
+}
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('CloseContact', () => {
+  it('Properly renders all main components for empty close contact', () => {
+    const emptyCCWrapper = getShallowWrapper(mockCloseContact1);
+    expect(emptyCCWrapper.find('div').at(0).text()).toContain('Add New Close Contact');
+
+    expect(emptyCCWrapper.state('showModal')).toBeFalsy();
+    expect(emptyCCWrapper.state('first_name')).toEqual('')
+    expect(emptyCCWrapper.state('last_name')).toEqual('')
+    expect(emptyCCWrapper.state('primary_telephone')).toEqual('')
+    expect(emptyCCWrapper.state('email')).toEqual('')
+    expect(emptyCCWrapper.state('last_date_of_exposure')).toEqual(null)
+    expect(emptyCCWrapper.state('assigned_user')).toEqual(null)
+    expect(emptyCCWrapper.state('notes')).toEqual('')
+    expect(emptyCCWrapper.state('enrolled_id')).toEqual(null)
+    expect(emptyCCWrapper.state('contact_attempts')).toEqual(0)
+  });
+
+  it('Properly renders all main components for already existing close contact', () => {
+    const alreadyExistingCCWrapper = getShallowWrapper(mockCloseContact2);
+    expect(alreadyExistingCCWrapper.find('div').at(0).text()).toContain('Edit Contact Attempt');
```

Comment:
I feel like a clearer way to test this is to grab the individual buttons - this makes it seem like there's an "Edit Contact Attempt" button at first.

Reply:
Agreed. Changing with the latest commit.